File: app/src/main/java/com/chemecador/secretaria/api/ApiException.java
package com.chemecador.secretaria.api;

import java.io.IOException;

public class ApiException extends IOException {

    private final int responseCode;

    public ApiException(int responseCode, String message) {
        super(message);
        this.responseCode = responseCode;
    }

    public ApiException(int responseCode) {
        this.responseCode = responseCode;
    }

    public int getResponseCode() {
        return responseCode;
    }
}

File: app/src/main/java/com/chemecador/secretaria/api/ApiException.kt
package com.chemecador.secretaria.api

import java.io.IOException

class ApiException : IOException {
    private val responseCode: Int

    constructor(responseCode: Int, message: String?) : super(message) {
        this.responseCode = responseCode
    }

    constructor(responseCode: Int) {
        this.responseCode = responseCode
    }
}